src/session/session_api.c

```diff
@@ -1449,7 +1456,44 @@ __wt_session_range_truncate(
             goto done;
     }
 
-    WT_ERR(__wt_schema_range_truncate(session, start, stop));
+    /*
+     * Create a copy of the start and stop cursors to maintain the original start and stop positions
+     * for error-checking purposes.
+     */
+#ifdef HAVE_DIAGNOSTIC
+    debug_start = debug_stop = NULL;
+    if (start != NULL)
+        WT_ERR(__session_open_cursor((WT_SESSION *)session, NULL, start, NULL, &debug_start));
+    if (stop != NULL)
+        WT_ERR(__session_open_cursor((WT_SESSION *)session, NULL, stop, NULL, &debug_stop));
+#endif
+
+    WT_ERR(__wt_schema_range_truncate(session, start, stop, &is_col_fix));
+
+#ifdef HAVE_DIAGNOSTIC
+    /*
+     * The debug cursors will be positioned at the start and stop keys of the range if there is one.
+     * For row-store and variable-length column store, we expect a WT_NOTFOUND value when searching
+     * for a record that has been truncated. For fixed length column store, this works a little
+     * differently. We should instead check that the corresponding value of the truncated record is
+     * zero.
+     */
+    if (!is_col_fix) {
+        if (start != NULL)
+            WT_ASSERT(session, debug_start->search(debug_start) == WT_NOTFOUND);
+        if (stop != NULL)
+            WT_ASSERT(session, debug_stop->search(debug_stop) == WT_NOTFOUND);
+    } else {
+        if (start != NULL) {
+            debug_start->search(debug_start);
```

Comment:
Should wrap the `search` calls around `WT_ERR`?

Reply:
Good idea, yes I agree we need to use `WT_ERR` as the cursors need to be cleaned up on failure.